keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/proposal/AcceptanceRateTracker.java

```diff
@@ -0,0 +1,34 @@
+package io.improbable.keanu.algorithms.mcmc.proposal;
+
+import com.google.common.collect.Maps;
+import io.improbable.keanu.vertices.Vertex;
+
+import java.util.Map;
+import java.util.Set;
+
+public class AcceptanceRateTracker implements ProposalListener {
+
+    private Map<Set<Vertex>, Integer> numApplied = Maps.newHashMap();
+    private Map<Set<Vertex>, Integer> numRejected = Maps.newHashMap();
+
+    @Override
+    public void onProposalApplied(Proposal proposal) {
+        Set<Vertex> key = proposal.getVerticesWithProposal();
+        Integer previousValue = numApplied.getOrDefault(key, 0);
+        numApplied.put(key, previousValue + 1);
+    }
+
+    public double getAcceptanceRate(Set<Vertex> verticesWithProposal) {
```

Comment:
As a user - how am I supposed to know the correct "Set" to pass here?  Might this not be dependent on the Proposal distribution in use?

Reply:
The user sets the type of `MHStepVariableSelector`. There are only two options currently:
1. select the variables one by one individually
2. select all the variables

So I _think_ this is fine, since the options are very simple and the user has control over the choice. It's possible that this may become more complicated in the future, although at present we always use option 1 so I don't think there's any demand for more choices.

I'll ask some of the data scientists for their opinion.

Reply:
I asked Charlie, and he said that the other possible option would be for the user to define a partition of their latent vertices. The existing 2 options could be thought of as special cases of this partition.

So whichever of these 3 options  we choose, it's a partition, i.e. a given vertex will exist in strictly 1 set. Therefore I agree with you - we can use individual vertices for the key rather than sets.

I've implemented it to key on `VertexId`.